src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/TempMultipleAlignmentsReclassifier.java

```diff
@@ -0,0 +1,211 @@
+
+        // 2nd and 4th classes
+        final JavaPairRDD<AlignedContig, List<String>> multipleAlignments =
+                split1._2.mapToPair(tuple3 -> new Tuple2<>(tuple3._2(), tuple3._3()));
+        final JavaPairRDD<AlignedContig, List<String>> multipleAlignmentsIncompletePicture =
+                multipleAlignments.filter(pair -> hasIncomePicture(pair._1));
+        final JavaPairRDD<AlignedContig, List<String>> multipleAlignmentsCompletePicture =
+                multipleAlignments.filter(pair -> !hasIncomePicture(pair._1));
+
+        return new Tuple4<>(garbage, multipleAlignmentsIncompletePicture.keys(),
+                twoAlignments, multipleAlignmentsCompletePicture);
+    }
+
+    private static boolean hasIncomePicture(final AlignedContig contigWithMultipleAlignments) {
+        Utils.validateArg(contigWithMultipleAlignments.alignmentIntervals.size() > 2,
+                "assumption that input contig has more than 2 alignments is violated.\n" +
+                        contigWithMultipleAlignments.toString());
+
+        final AlignmentInterval head = contigWithMultipleAlignments.alignmentIntervals.get(0),
+                                tail = contigWithMultipleAlignments.alignmentIntervals.get(contigWithMultipleAlignments.alignmentIntervals.size()-1);
+
+        // tail not resuming the "direction of flow" started by head
+        if ( !head.referenceSpan.getContig().equals(tail.referenceSpan.getContig()) || head.forwardStrand != tail.forwardStrand)
+            return true;
+
+        // reference span anomaly
+        final SimpleInterval referenceSpanHead = head.referenceSpan,
+                             referenceSpanTail = tail.referenceSpan;
+        if (referenceSpanHead.contains(referenceSpanTail) || referenceSpanTail.contains(referenceSpanHead))
+            return true;
+
+        final boolean refSpanAnomaly =
+                contigWithMultipleAlignments.alignmentIntervals
+                        .subList(1, contigWithMultipleAlignments.alignmentIntervals.size() - 1)
+                        .stream().map(ai -> ai.referenceSpan)
+                        .anyMatch( middle -> { // middle alignments' ref span either disjoint from head/tail, or completely contained in head/tail ref span
+                            final boolean badHead = middle.overlaps(referenceSpanHead) && !referenceSpanHead.contains(middle);
+                            final boolean badTail = middle.overlaps(referenceSpanTail) && !referenceSpanTail.contains(middle);
+                            return badHead || badTail;
+                        });
+        if (refSpanAnomaly)
+            return true;
+
+        if (head.forwardStrand) {
+            return referenceSpanHead.getStart() >= referenceSpanTail.getStart();
+        } else {
+            return referenceSpanHead.getEnd() <= referenceSpanHead.getEnd();
+        }
+    }
+
+    private static boolean isNonInformative(final Tuple3<Boolean, AlignedContig, List<String>> tigInfo) {
```

Comment:
I'd just make this `Tuple3` into a little class, then you don't need these extra methods to decode their entries.